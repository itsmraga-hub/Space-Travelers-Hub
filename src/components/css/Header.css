@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap');

body {
  overflow-x: hidden;
}

nav {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
}

.logo-container {
  display: flex;
  width: 50%;
  align-items: center;
}

.logo-container a {
  text-decoration: none;
  font-family: 'Lato', sans-serif;
  font-size: 2rem;
  color: #000;
}

.logo {
  width: 50px;
  height: 50px;
  margin-right: 1rem;
}

.nav-link-container {
  width: 50%;
  display: flex;
  justify-content: space-around;
  list-style: none;
}

.nav-link {
  font-family: 'Lato', sans-serif;
  font-size: 1.2rem;
  font-weight: 400;
}

.link {
  text-decoration: underline;
}

.active {
  font-weight: 700;
  color: blue;
}
