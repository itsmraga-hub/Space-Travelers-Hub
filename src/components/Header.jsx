import './Header.css';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <nav>
      <div className='logo-container'>
        <img src="/images/planet.png" className="logo" alt="logo" />
        <NavLink to="/rockets" className="link">Space Travelers' Hub</NavLink>
      </div>
      <ul>
        <li><NavLink to="/rockets" className="link" activeClassName="active">Rockets</NavLink></li>
        <li><NavLink to="/missions" className="link" activeClassName="active">Missions</NavLink></li>
        <li><NavLink to="/my-profile" className="link" activeClassName="active">My Profile</NavLink></li>
      </ul>
    </nav>
  );


export default Header;
