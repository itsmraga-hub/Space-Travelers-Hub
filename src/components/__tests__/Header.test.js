import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Header from '../Header';

it('<Header /> renders properly as expected', () => {
  const tree = renderer.create(<BrowserRouter><Header /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
