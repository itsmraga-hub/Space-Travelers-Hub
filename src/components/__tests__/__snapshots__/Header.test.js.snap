// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders properly as expected 1`] = `
<header>
  <nav>
    <div
      className="logo-container"
    >
      <img
        alt="logo"
        className="logo"
        src="/images/planet.png"
      />
      <a
        className="brand-logo"
        href="/"
        onClick={[Function]}
      >
        Space Travelers' Hub
      </a>
    </div>
    <ul
      className="nav-link-container"
    >
      <li
        className="nav-link"
      >
        <a
          activeclassname="active"
          aria-current="page"
          className="link active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          activeclassname="active"
          className="link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          activeclassname="active"
          className="link"
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
